src/components/Modal/BaseModal.tsx

```diff
@@ -141,6 +141,7 @@ function BaseModal(
     };
 
     const handleDismissModal = () => {
+        hideModal();
```

Comment:
@tsa321 I can't reproduce https://github.com/Expensify/App/issues/27237 on the latest main. Should we revert this change?

Reply:
Yes, I will revert BaseModal file

Reply:
@DylanDylann I cannot reproduce it in staging, but I can reproduce it in latest main. 
It is even more broken in latest main, even opening receipt modal -> cmd +`k` won't open search page. Could you check it?

Also my code depends on the `hideModal` function inside BaseModal, but when closing  receipt modal the `hideModal` function is not executed. So I make that modification in BaseModal. 
Could you check it thank you...

Reply:
@tsa321 Try again and still can't reproduce on the latest main

Reply:
Anyway, I think it is out of the scope of this PR, we should revert that to avoid delay processing

Reply:
@DylanDylann alright. I will make minimal changes. 
If I make modification in the beforeRemove I need the changes in baseModal, so the composer will re-focus after user close receipt modal.
But I will move the condition in shouldAutoFocus of ComposerWithSuggestions and revert  the changes in BaseModal and AuthScreen. So the changes will be minimal.